graylog2-server/src/main/java/org/graylog/events/processor/aggregation/AggregationEventProcessor.java

```diff
@@ -236,12 +244,32 @@ private void filterSearch(EventFactory eventFactory, AggregationEventProcessorPa
                         .build());
 
                 eventsWithContext.add(EventWithContext.create(event, msg));
+                if (config.eventLimit() != 0) {
+                    if (messageCount.incrementAndGet() >= config.eventLimit()) {
+                        eventLimitReached = true;
```

Comment:
why not just `throw new EventLimitReachedException();;` directly without the boolean?